drivers/audio/rk3xi2sbus/rk-tplg.cpp

```diff
@@ -0,0 +1,164 @@
+#include "driver.h"
+#include <acpiioct.h>
+#include "rk-tplg.h"
+
+DEFINE_GUID(GUID_ACPI_DSD,
+	0xdaffd814, 0x6eba, 0x4d8c, 0x8a, 0x91, 0xbc, 0x9b, 0xbf, 0x4a, 0xa3, 0x01);
+
+void copyDSDParam(PACPI_METHOD_ARGUMENT dsdParameterData, char** buf) {
+	RtlZeroMemory(buf, 32);
+	RtlCopyMemory(buf, dsdParameterData->Data, min(dsdParameterData->DataLength, 32));
+}
+
+void copyDSDParamNum(PACPI_METHOD_ARGUMENT dsdParameterData, UINT32* buf) {
+	RtlZeroMemory(buf, sizeof(*buf));
+	RtlCopyMemory(buf, dsdParameterData->Data, min(dsdParameterData->DataLength, sizeof(*buf)));
+}
+
+NTSTATUS
+GetRKTplg(
+	_In_ WDFDEVICE FxDevice,
+	RK_TPLG* rkTplg
+)
+{
+	if (!rkTplg) {
+		return STATUS_INVALID_PARAMETER;
+	}
+
+	NTSTATUS status = STATUS_ACPI_NOT_INITIALIZED;
+	ACPI_EVAL_INPUT_BUFFER_EX inputBuffer;
+	RtlZeroMemory(&inputBuffer, sizeof(inputBuffer));
+
+	inputBuffer.Signature = ACPI_EVAL_INPUT_BUFFER_SIGNATURE_EX;
+	status = RtlStringCchPrintfA(
```

Comment:
Since there aren't any format inserts in `"_DSD"`, and since you're using `sizeof` for the buffer size, Consider `RtlStringCbCopyA`.

As a matter of taste, since `RtlStringCbCopyA` fails in a generally-safe mode (it truncates the copy and nul-terminates the output buffer), I tend to feel it's ok to ignore the return code from it. The only way it can fail is if you passed invalid parameters (which you don't), and it fails in a safe way even if you do (truncates the output), so checking the result doesn't do much good and just makes the code longer.